networkmanager: drop hard dependency on openconnect and cleanup plugin handling

mweinelt opened the pull request:
Its closure size is huge, due to a dependency on webkitgtk, and now that
we want to enable networkmanager on the ISOs by default closure size
matters a lot.

Instead we now pass openconnect to the PATH of the systemd unit and pick
it up from there.

<del>:warning: I did not test the openconnect plugin, not sure I can reasonably do that.</del>
:heavy_check_mark:  I tested an openconnect connection and it worked the same as before with the patch applied.

---

Since opening this PR I also redid the plugin requirement handling and dropped options that I felt were a mistake. Thanks for the inspiration and review by the #network:nixos.org folks.

---

cc #420923 

<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [Nixpkgs 25.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/doc/release-notes/rl-2511.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/doc/manual/release-notes/rl-2505.section.md) Nixpkgs Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
- [NixOS 25.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2511.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) NixOS Release notes)
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md), [pkgs/README.md](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md), [maintainers/README.md](https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md) and other contributing documentation in corresponding paths.

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

mweinelt committed:
networkmanager: drop hard dependency on openconnect

Its closure size is huge, due to a dependency on webkitgtk, and now that
we want to enable networkmanager on the ISOs by default closure size
matters a lot.

Instead we now pass openconnect to the PATH of the systemd unit and pick
it up from there.

mweinelt committed:
nixos/networkmanager: clean up plugin handling

There is no point in having a special option to enable strongswan, when
we can just parse the intent from the plugin list instead.

Also pick up relevant runtime dependency information from the plugin
package instead of providing additional options or hardcoding them.

SuperSandro2000 commented on nixos/modules/services/networking/networkmanager.nix:
```diff
@@ -523,6 +512,9 @@ in
       [ "networking" "modemmanager" "fccUnlockScripts" ]
     )
+    (mkRemovedOptionModule [ "network" "networkmanager" "enableStrongSwan" ]
+      "This option was removed. Pass `pkgs.networkmanager_strongswan` into `networking.networkmanager.plugins` instead."
```
```suggestion
      "Pass `pkgs.networkmanager_strongswan` into `networking.networkmanager.plugins` instead."
```
Otherwise that part would duplicated https://github.com/NixOS/nixpkgs/blob/dcc4da22e1bf5ed04220e7c5ddaa80abc0dee646/lib/modules.nix#L1497

SuperSandro2000 commented:
I've been running without ModemManager for a while. Not sure if that can be as easy removed, too.

```nix
  networkmanager = (prev.networkmanager.override {
    modemmanager = pkgs.emptyDirectory;
    openconnect = pkgs.emptyDirectory;
  }).overrideAttrs ({ mesonFlags, ... }: {
    mesonFlags = mesonFlags ++ [
      "-Dmodem_manager=false"
    ];
  });
```

mweinelt commented:
We've discussed modemmanager, and it is a less interesting target for closure size reduction. But in the long run most of the dependencies should be made optional runtime dependencies.

mweinelt commented:
Migrated all plugin runtime requirements into the plugins passthru attribute set, so we can just pick them up from there without matching the pname or anything unstable like it.

mweinelt committed:
nixos/networkmanager: drop default plugin list and toggle

It is unclear where this list originated, but it doesn't make sense to
ship it with all networkmanager installations. The most excessive plugin
is openconnect, that ships a 250 MB closure including webkitgtk.

Instead users now have to specify the plugins they want explicitly. I
updated the option to give hints on how to find them as best as I can.

emilazy reviewed (approved):
I think this is a good idea, we cannot ship every plug‐in under the sun by default. The error for connections that will break as a result of this seems good (naming the plug‐in explicitly). I am not sure if there is more we can do for migration. We could move his default into the GNOME/Plasma configurations or something, but I don’t know if it belongs there. It’d be good to have some idea of what plug‐ins other distros ship out of the box.

emilazy commented:
Uh but the manual build should probably be fixed :)

mweinelt committed:
networkmanager-strongswan: rename from networkmanager_strongswan

This is to follow the same structure as all other nm plugins.

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/breaking-changes-announcement-for-unstable/17574/85

jtojnar reviewed (approved):
Nice cleanup.

LGTM. Also successfully connected using openconnect VPN.

misuzu commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 421042`
Commit: `312015eaafbc3af80571bd68c465a6929cf710de`

---
### `x86_64-linux`
<details>
  <summary>:fast_forward: 4 packages marked as broken and skipped:</summary>
  <ul>
    <li>kdePackages.itinerary</li>
    <li>kdePackages.itinerary.debug</li>
    <li>kdePackages.itinerary.dev</li>
    <li>kdePackages.itinerary.devtools</li>
  </ul>
</details>
<details>
  <summary>:fast_forward: 2 packages blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
    <li>tests.nixos-functions.nixos-test</li>
  </ul>
</details>
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>libsForQt5.kdev-python (plasma5Packages.kdev-python)</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 634 packages built:</summary>
  <ul>
    <li>CuboCore.coretoppings</li>
    <li>adapta-gtk-theme</li>
    <li>ags_1</li>
    <li>airgeddon</li>
    <li>appgate-sdp</li>
    <li>arma3-unix-launcher</li>
    <li>astal.network</li>
    <li>autosuspend</li>
    <li>autosuspend.dist</li>
    <li>ayatana-indicator-bluetooth</li>
    <li>ayatana-indicator-datetime</li>
    <li>ayatana-indicator-display</li>
    <li>ayatana-indicator-messages</li>
    <li>ayatana-indicator-messages.dev</li>
    <li>ayatana-indicator-messages.devdoc</li>
    <li>ayatana-indicator-power</li>
    <li>ayatana-indicator-session</li>
    <li>ayatana-indicator-sound</li>
    <li>better-control</li>
    <li>blueman</li>
    <li>bottles</li>
    <li>bottles-unwrapped</li>
    <li>breath-theme</li>
    <li>bs-manager</li>
    <li>budgie-analogue-clock-applet</li>
    <li>budgie-control-center</li>
    <li>budgie-control-center.debug</li>
    <li>budgie-desktop</li>
    <li>budgie-desktop-with-plugins</li>
    <li>budgie-desktop.dev</li>
    <li>budgie-desktop.man</li>
    <li>budgie-gsettings-overrides</li>
    <li>budgie-session</li>
    <li>budgie-session.debug</li>
    <li>budgie-session.man</li>
    <li>budgie-systemmonitor-applet</li>
    <li>budgie-user-indicator-redux</li>
    <li>cage</li>
    <li>cagebreak</li>
    <li>cambalache</li>
    <li>casilda</li>
    <li>casilda.dev</li>
    <li>cinnamon-common</li>
    <li>cinnamon-control-center</li>
    <li>cinnamon-gsettings-overrides</li>
    <li>cinnamon-screensaver</li>
    <li>cinnamon-session</li>
    <li>cinnamon-settings-daemon</li>
    <li>cinnamon-settings-daemon.dev</li>
    <li>claws-mail</li>
    <li>claws-mail.dev</li>
    <li>clight</li>
    <li>cobang</li>
    <li>code-nautilus</li>
    <li>collision</li>
    <li>colloid-kde</li>
    <li>colloid-kde.sddm</li>
    <li>dbmonster</li>
    <li>dbus-test-runner</li>
    <li>deepin.dde-daemon</li>
    <li>deepin.dde-file-manager</li>
    <li>deepin.dde-grand-search</li>
    <li>deepin.dde-gsettings-schemas</li>
    <li>deepin.dde-launchpad</li>
    <li>deepin.dde-network-core</li>
    <li>deepin.dde-session-ui</li>
    <li>deepin.dde-shell</li>
    <li>deepin.dde-tray-loader</li>
    <li>deepin.deepin-system-monitor</li>
    <li>deltatouch</li>
    <li>deskflow</li>
    <li>dropbox-cli</li>
    <li>dropbox-cli.nautilusExtension</li>
    <li>dwl</li>
    <li>dwl.man</li>
    <li>easyroam-connect-desktop</li>
    <li>eduvpn-client</li>
    <li>eduvpn-client.dist</li>
    <li>eiciel</li>
    <li>eiciel.nautilusExtension</li>
    <li>eloquent</li>
    <li>file-roller</li>
    <li>firewalld</li>
    <li>firewalld-gui</li>
    <li>fprintd-tod</li>
    <li>fprintd-tod.devdoc</li>
    <li>fractal</li>
    <li>gamescope</li>
    <li>gdm</li>
    <li>gdm.debug</li>
    <li>gdm.dev</li>
    <li>gnome-applets</li>
    <li>gnome-bluetooth</li>
    <li>gnome-bluetooth.dev</li>
    <li>gnome-bluetooth.devdoc</li>
    <li>gnome-bluetooth.man</li>
    <li>gnome-browser-connector</li>
    <li>gnome-control-center</li>
    <li>gnome-control-center.debug</li>
    <li>gnome-disk-utility</li>
    <li>gnome-flashback</li>
    <li>gnome-initial-setup</li>
    <li>gnome-network-displays</li>
    <li>gnome-panel</li>
    <li>gnome-panel-with-modules</li>
    <li>gnome-panel.dev</li>
    <li>gnome-panel.man</li>
    <li>gnome-photos</li>
    <li>gnome-photos.installedTests</li>
    <li>gnome-power-manager</li>
    <li>gnome-remote-desktop</li>
    <li>gnome-session</li>
    <li>gnome-session.debug</li>
    <li>gnome-session.sessions</li>
    <li>gnome-settings-daemon</li>
    <li>gnome-settings-daemon46 (pantheon.gnome-settings-daemon)</li>
    <li>gnome-shell</li>
    <li>gnome-shell.debug</li>
    <li>gnome-shell.devdoc</li>
    <li>gnome-terminal</li>
    <li>gnome-tweaks</li>
    <li>gnome-usage</li>
    <li>gnome.nixos-gsettings-overrides</li>
    <li>gnomeExtensions.easyScreenCast</li>
    <li>gnomeExtensions.gsconnect</li>
    <li>gnomeExtensions.gsconnect.installedTests</li>
    <li>gnomeExtensions.gtk4-desktop-icons-ng-ding</li>
    <li>graphite-kde-theme</li>
    <li>grimblast</li>
    <li>haskellPackages.gi-nm</li>
    <li>haskellPackages.gi-nm.doc</li>
    <li>hdrop</li>
    <li>hyprland</li>
    <li>hyprland.dev</li>
    <li>hyprland.man</li>
    <li>hyprlandPlugins.borders-plus-plus</li>
    <li>hyprlandPlugins.csgo-vulkan-fix</li>
    <li>hyprlandPlugins.hy3</li>
    <li>hyprlandPlugins.hypr-dynamic-cursors</li>
    <li>hyprlandPlugins.hyprbars</li>
    <li>hyprlandPlugins.hyprexpo</li>
    <li>hyprlandPlugins.hyprgrass</li>
    <li>hyprlandPlugins.hyprscrolling</li>
    <li>hyprlandPlugins.hyprspace</li>
    <li>hyprlandPlugins.hyprsplit</li>
    <li>hyprlandPlugins.hyprtrails</li>
    <li>hyprlandPlugins.hyprwinwrap</li>
    <li>hyprlandPlugins.xtra-dispatchers</li>
    <li>hyprpanel</li>
    <li>hyprshade</li>
    <li>hyprshade.dist</li>
    <li>hyprshot</li>
    <li>ifwifi</li>
    <li>input-leap</li>
    <li>insync-nautilus</li>
    <li>itch</li>
    <li>jami</li>
    <li>kaffeine</li>
    <li>kde-rounded-corners</li>
    <li>kdePackages.dynamic-workspaces</li>
    <li>kdePackages.karousel</li>
    <li>kdePackages.kdeplasma-addons</li>
    <li>kdePackages.kdeplasma-addons.debug</li>
    <li>kdePackages.kdeplasma-addons.dev</li>
    <li>kdePackages.kdeplasma-addons.devtools</li>
    <li>kdePackages.koi</li>
    <li>kdePackages.kongress</li>
    <li>kdePackages.kongress.debug</li>
    <li>kdePackages.kongress.dev</li>
    <li>kdePackages.kongress.devtools</li>
    <li>kdePackages.kosmindoormap</li>
    <li>kdePackages.kosmindoormap.debug</li>
    <li>kdePackages.kosmindoormap.dev</li>
    <li>kdePackages.kosmindoormap.devtools</li>
    <li>kdePackages.kpublictransport</li>
    <li>kdePackages.kpublictransport.debug</li>
    <li>kdePackages.kpublictransport.dev</li>
    <li>kdePackages.kpublictransport.devtools</li>
    <li>kdePackages.krohnkite</li>
    <li>kdePackages.ksystemstats</li>
    <li>kdePackages.ksystemstats.debug</li>
    <li>kdePackages.ksystemstats.dev</li>
    <li>kdePackages.ksystemstats.devtools</li>
    <li>kdePackages.ktrip</li>
    <li>kdePackages.ktrip.debug</li>
    <li>kdePackages.ktrip.dev</li>
    <li>kdePackages.ktrip.devtools</li>
    <li>kdePackages.kup</li>
    <li>kdePackages.kup.debug</li>
    <li>kdePackages.kup.dev</li>
    <li>kdePackages.kup.devtools</li>
    <li>kdePackages.kwin</li>
    <li>kdePackages.kwin-x11</li>
    <li>kdePackages.kwin-x11.debug</li>
    <li>kdePackages.kwin-x11.dev</li>
    <li>kdePackages.kwin-x11.devtools</li>
    <li>kdePackages.kwin.debug</li>
    <li>kdePackages.kwin.dev</li>
    <li>kdePackages.kwin.devtools</li>
    <li>kdePackages.kzones</li>
    <li>kdePackages.networkmanager-qt</li>
    <li>kdePackages.networkmanager-qt.debug</li>
    <li>kdePackages.networkmanager-qt.dev</li>
    <li>kdePackages.networkmanager-qt.devtools</li>
    <li>kdePackages.plasma-browser-integration</li>
    <li>kdePackages.plasma-browser-integration.debug</li>
    <li>kdePackages.plasma-browser-integration.dev</li>
    <li>kdePackages.plasma-browser-integration.devtools</li>
    <li>kdePackages.plasma-desktop</li>
    <li>kdePackages.plasma-desktop.debug</li>
    <li>kdePackages.plasma-desktop.dev</li>
    <li>kdePackages.plasma-desktop.devtools</li>
    <li>kdePackages.plasma-mobile</li>
    <li>kdePackages.plasma-mobile.debug</li>
    <li>kdePackages.plasma-mobile.dev</li>
    <li>kdePackages.plasma-mobile.devtools</li>
    <li>kdePackages.plasma-nm</li>
    <li>kdePackages.plasma-nm.debug</li>
    <li>kdePackages.plasma-nm.dev</li>
    <li>kdePackages.plasma-nm.devtools</li>
    <li>kdePackages.plasma-pa</li>
    <li>kdePackages.plasma-pa.debug</li>
    <li>kdePackages.plasma-pa.dev</li>
    <li>kdePackages.plasma-pa.devtools</li>
    <li>kdePackages.plasma-sdk</li>
    <li>kdePackages.plasma-sdk.debug</li>
    <li>kdePackages.plasma-sdk.dev</li>
    <li>kdePackages.plasma-sdk.devtools</li>
    <li>kdePackages.plasma-vault</li>
    <li>kdePackages.plasma-vault.debug</li>
    <li>kdePackages.plasma-vault.dev</li>
    <li>kdePackages.plasma-vault.devtools</li>
    <li>kdePackages.plasma-workspace</li>
    <li>kdePackages.plasma-workspace.debug</li>
    <li>kdePackages.plasma-workspace.dev</li>
    <li>kdePackages.plasma-workspace.devtools</li>
    <li>kdePackages.plasma5support</li>
    <li>kdePackages.plasma5support.debug</li>
    <li>kdePackages.plasma5support.dev</li>
    <li>kdePackages.plasma5support.devtools</li>
    <li>kdePackages.powerdevil</li>
    <li>kdePackages.powerdevil.debug</li>
    <li>kdePackages.powerdevil.dev</li>
    <li>kdePackages.powerdevil.devtools</li>
    <li>kdePackages.qrca</li>
    <li>kdePackages.qrca.debug</li>
    <li>kdePackages.qrca.dev</li>
    <li>kdePackages.qrca.devtools</li>
    <li>kdePackages.qwlroots (qt6Packages.qwlroots)</li>
    <li>kdePackages.sierra-breeze-enhanced (qt6Packages.sierra-breeze-enhanced)</li>
    <li>kdePackages.wacomtablet</li>
    <li>kdePackages.wacomtablet.debug</li>
    <li>kdePackages.wacomtablet.dev</li>
    <li>kdePackages.wacomtablet.devtools</li>
    <li>kdePackages.waylib (qt6Packages.waylib)</li>
    <li>kdePackages.waylib.dev (qt6Packages.waylib.dev)</li>
    <li>kmplayer</li>
    <li>kmymoney</li>
    <li>kodiPackages.steam-launcher</li>
    <li>labwc</li>
    <li>labwc.doc</li>
    <li>labwc.man</li>
    <li>latte-dock</li>
    <li>layan-kde</li>
    <li>libayatana-common</li>
    <li>libei</li>
    <li>libnma</li>
    <li>libnma-gtk4</li>
    <li>libnma-gtk4.dev</li>
    <li>libnma-gtk4.devdoc</li>
    <li>libnma.dev</li>
    <li>libnma.devdoc</li>
    <li>libqtdbusmock</li>
    <li>libqtdbustest</li>
    <li>libsForQt5.accounts-qml-module (plasma5Packages.accounts-qml-module)</li>
    <li>libsForQt5.accounts-qml-module.doc (plasma5Packages.accounts-qml-module.doc)</li>
    <li>libsForQt5.alkimia (plasma5Packages.alkimia)</li>
    <li>libsForQt5.applet-window-appmenu (plasma5Packages.applet-window-appmenu)</li>
    <li>libsForQt5.bismuth (plasma5Packages.bismuth)</li>
    <li>libsForQt5.kasts (plasma5Packages.kasts)</li>
    <li>libsForQt5.kde-cli-tools (plasma5Packages.kde-cli-tools)</li>
    <li>libsForQt5.kdelibs4support (plasma5Packages.kdelibs4support)</li>
    <li>libsForQt5.kdelibs4support.dev (plasma5Packages.kdelibs4support.dev)</li>
    <li>libsForQt5.kdeplasma-addons (plasma5Packages.kdeplasma-addons)</li>
    <li>libsForQt5.kdev-php (plasma5Packages.kdev-php)</li>
    <li>libsForQt5.kdevelop (plasma5Packages.kdevelop)</li>
    <li>libsForQt5.kdevelop-unwrapped (plasma5Packages.kdevelop-unwrapped)</li>
    <li>libsForQt5.kget (plasma5Packages.kget)</li>
    <li>libsForQt5.khotkeys (plasma5Packages.khotkeys)</li>
    <li>libsForQt5.khotkeys.bin (plasma5Packages.khotkeys.bin)</li>
    <li>libsForQt5.khotkeys.dev (plasma5Packages.khotkeys.dev)</li>
    <li>libsForQt5.kinfocenter (plasma5Packages.kinfocenter)</li>
    <li>libsForQt5.kmenuedit (plasma5Packages.kmenuedit)</li>
    <li>libsForQt5.kosmindoormap (plasma5Packages.kosmindoormap)</li>
    <li>libsForQt5.kosmindoormap.dev (plasma5Packages.kosmindoormap.dev)</li>
    <li>libsForQt5.kpublictransport (plasma5Packages.kpublictransport)</li>
    <li>libsForQt5.krohnkite (plasma5Packages.krohnkite)</li>
    <li>libsForQt5.krunner-symbols (plasma5Packages.krunner-symbols)</li>
    <li>libsForQt5.ksystemstats (plasma5Packages.ksystemstats)</li>
    <li>libsForQt5.ktrip (plasma5Packages.ktrip)</li>
    <li>libsForQt5.kwin (plasma5Packages.kwin)</li>
    <li>libsForQt5.kwin-dynamic-workspaces (plasma5Packages.kwin-dynamic-workspaces)</li>
    <li>libsForQt5.kwin-tiling (plasma5Packages.kwin-tiling)</li>
    <li>libsForQt5.kwin.dev (plasma5Packages.kwin.dev)</li>
    <li>libsForQt5.kzones (plasma5Packages.kzones)</li>
    <li>libsForQt5.networkmanager-qt (plasma5Packages.networkmanager-qt)</li>
    <li>libsForQt5.networkmanager-qt.dev (plasma5Packages.networkmanager-qt.dev)</li>
    <li>libsForQt5.parachute (plasma5Packages.parachute)</li>
    <li>libsForQt5.plasma-bigscreen (plasma5Packages.plasma-bigscreen)</li>
    <li>libsForQt5.plasma-browser-integration (plasma5Packages.plasma-browser-integration)</li>
    <li>libsForQt5.plasma-desktop (plasma5Packages.plasma-desktop)</li>
    <li>libsForQt5.plasma-disks (plasma5Packages.plasma-disks)</li>
    <li>libsForQt5.plasma-disks.bin (plasma5Packages.plasma-disks.bin)</li>
    <li>libsForQt5.plasma-disks.dev (plasma5Packages.plasma-disks.dev)</li>
    <li>libsForQt5.plasma-mobile (plasma5Packages.plasma-mobile)</li>
    <li>libsForQt5.plasma-nm (plasma5Packages.plasma-nm)</li>
    <li>libsForQt5.plasma-remotecontrollers (plasma5Packages.plasma-remotecontrollers)</li>
    <li>libsForQt5.plasma-settings (plasma5Packages.plasma-settings)</li>
    <li>libsForQt5.plasma-systemmonitor (plasma5Packages.plasma-systemmonitor)</li>
    <li>libsForQt5.plasma-vault (plasma5Packages.plasma-vault)</li>
    <li>libsForQt5.plasma-workspace (plasma5Packages.plasma-workspace)</li>
    <li>libsForQt5.plasma-workspace.dev (plasma5Packages.plasma-workspace.dev)</li>
    <li>libsForQt5.powerdevil (plasma5Packages.powerdevil)</li>
    <li>libsForQt5.qtcurve (plasma5Packages.qtcurve)</li>
    <li>sierra-breeze-enhanced (libsForQt5.sierra-breeze-enhanced)</li>
    <li>soundkonverter (libsForQt5.soundkonverter)</li>
    <li>libsForQt5.systemsettings (plasma5Packages.systemsettings)</li>
    <li>libsForQt5.systemsettings.bin (plasma5Packages.systemsettings.bin)</li>
    <li>libsForQt5.systemsettings.dev (plasma5Packages.systemsettings.dev)</li>
    <li>libsForQt5.umbrello (plasma5Packages.umbrello)</li>
    <li>libsForQt5.xdg-desktop-portal-kde (plasma5Packages.xdg-desktop-portal-kde)</li>
    <li>lightly-boehs</li>
    <li>lightly-qt</li>
    <li>linux-router</li>
    <li>linux-router-without-wifi</li>
    <li>linux-wifi-hotspot</li>
    <li>localsearch</li>
    <li>lomiri.gmenuharness</li>
    <li>lomiri.libusermetrics</li>
    <li>lomiri.libusermetrics.dev</li>
    <li>lomiri.libusermetrics.doc</li>
    <li>lomiri.lomiri</li>
    <li>lomiri.lomiri-action-api</li>
    <li>lomiri.lomiri-action-api.dev</li>
    <li>lomiri.lomiri-action-api.doc</li>
    <li>lomiri.lomiri-api</li>
    <li>lomiri.lomiri-api.dev</li>
    <li>lomiri.lomiri-api.doc</li>
    <li>lomiri.lomiri-app-launch</li>
    <li>lomiri.lomiri-app-launch.dev</li>
    <li>lomiri.lomiri-app-launch.doc</li>
    <li>lomiri.lomiri-calculator-app</li>
    <li>lomiri.lomiri-calendar-app</li>
    <li>lomiri.lomiri-camera-app</li>
    <li>lomiri.lomiri-clock-app</li>
    <li>lomiri.lomiri-content-hub</li>
    <li>lomiri.lomiri-content-hub.dev</li>
    <li>lomiri.lomiri-content-hub.doc</li>
    <li>lomiri.lomiri-content-hub.examples</li>
    <li>lomiri.lomiri-docviewer-app</li>
    <li>lomiri.lomiri-download-manager</li>
    <li>lomiri.lomiri-download-manager.dev</li>
    <li>lomiri.lomiri-download-manager.doc</li>
    <li>lomiri.lomiri-filemanager-app</li>
    <li>lomiri.lomiri-gallery-app</li>
    <li>lomiri.lomiri-history-service</li>
    <li>lomiri.lomiri-history-service.dev</li>
    <li>lomiri.lomiri-indicator-datetime</li>
    <li>lomiri.lomiri-indicator-network</li>
    <li>lomiri.lomiri-indicator-network.dev</li>
    <li>lomiri.lomiri-indicator-network.doc</li>
    <li>lomiri.lomiri-mediaplayer-app</li>
    <li>lomiri.lomiri-music-app</li>
    <li>lomiri.lomiri-notifications</li>
    <li>lomiri.lomiri-polkit-agent</li>
    <li>lomiri.lomiri-push-qml</li>
    <li>lomiri.lomiri-session</li>
    <li>lomiri.lomiri-system-settings</li>
    <li>lomiri.lomiri-system-settings-unwrapped</li>
    <li>lomiri.lomiri-system-settings-unwrapped.dev</li>
    <li>lomiri.lomiri-telephony-service</li>
    <li>lomiri.lomiri-terminal-app</li>
    <li>lomiri.lomiri-thumbnailer</li>
    <li>lomiri.lomiri-thumbnailer.dev</li>
    <li>lomiri.lomiri-thumbnailer.doc</li>
    <li>lomiri.lomiri-ui-extras</li>
    <li>lomiri.lomiri-ui-toolkit</li>
    <li>lomiri.lomiri-ui-toolkit.dev</li>
    <li>lomiri.lomiri-ui-toolkit.doc</li>
    <li>lomiri.lomiri-url-dispatcher</li>
    <li>lomiri.lomiri-url-dispatcher.dev</li>
    <li>lomiri.lomiri-url-dispatcher.lib</li>
    <li>lomiri.morph-browser</li>
    <li>lomiri.morph-browser.doc</li>
    <li>lomiri.qtmir</li>
    <li>lomiri.qtmir.dev</li>
    <li>lomiri.teleports</li>
    <li>lomiri.u1db-qt</li>
    <li>lomiri.u1db-qt.dev</li>
    <li>lomiri.u1db-qt.doc</li>
    <li>lomiri.u1db-qt.examples</li>
    <li>lutris</li>
    <li>magpie</li>
    <li>magpie.debug</li>
    <li>magpie.dev</li>
    <li>magpie.devdoc</li>
    <li>marble-shell-theme</li>
    <li>mate.mate-applets</li>
    <li>mate.mate-gsettings-overrides</li>
    <li>mate.mate-panel-with-applets</li>
    <li>mate.mate-power-manager</li>
    <li>mate.mate-tweak</li>
    <li>mate.mate-tweak.dist</li>
    <li>mate.mate-wayland-session</li>
    <li>matrix-gtk-theme</li>
    <li>minigalaxy</li>
    <li>minigalaxy.dist</li>
    <li>mir</li>
    <li>mir.dev</li>
    <li>mir_2_15</li>
    <li>mir_2_15.dev</li>
    <li>miracle-wm</li>
    <li>miriway</li>
    <li>mixxx</li>
    <li>mojave-gtk-theme</li>
    <li>monitor</li>
    <li>muffin</li>
    <li>muffin.dev</li>
    <li>muffin.man</li>
    <li>mutter</li>
    <li>mutter.debug</li>
    <li>mutter.dev</li>
    <li>mutter.devdoc</li>
    <li>mutter.man</li>
    <li>mutter46 (pantheon.mutter)</li>
    <li>mutter46.debug (pantheon.mutter.debug)</li>
    <li>mutter46.dev (pantheon.mutter.dev)</li>
    <li>mutter46.devdoc (pantheon.mutter.devdoc)</li>
    <li>mutter46.man (pantheon.mutter.man)</li>
    <li>nautilus</li>
    <li>nautilus-open-any-terminal</li>
    <li>nautilus-open-any-terminal.dist</li>
    <li>nautilus-open-in-blackbox</li>
    <li>nautilus-python</li>
    <li>nautilus-python.dev</li>
    <li>nautilus-python.devdoc</li>
    <li>nautilus-python.doc</li>
    <li>nautilus.dev</li>
    <li>nautilus.devdoc</li>
    <li>nemo-fileroller</li>
    <li>nemo-with-extensions</li>
    <li>nero-umu</li>
    <li>networkmanager</li>
    <li>networkmanager-fortisslvpn</li>
    <li>networkmanager-iodine</li>
    <li>networkmanager-l2tp</li>
    <li>networkmanager-openconnect</li>
    <li>networkmanager-openvpn</li>
    <li>networkmanager-sstp</li>
    <li>networkmanager-strongswan</li>
    <li>networkmanager-vpnc</li>
    <li>networkmanager.dev</li>
    <li>networkmanager.devdoc</li>
    <li>networkmanager.doc</li>
    <li>networkmanager.man</li>
    <li>networkmanager_dmenu</li>
    <li>networkmanagerapplet</li>
    <li>networkmanagerapplet.man</li>
    <li>nm-tray</li>
    <li>nordic</li>
    <li>nordic.sddm</li>
    <li>nwg-panel</li>
    <li>nwg-panel.dist</li>
    <li>opengamepadui</li>
    <li>pantheon-tweaks</li>
    <li>pantheon.elementary-greeter</li>
    <li>pantheon.elementary-gsettings-schemas</li>
    <li>pantheon.elementary-onboarding</li>
    <li>pantheon.elementary-session-settings</li>
    <li>pantheon.elementary-shortcut-overlay</li>
    <li>pantheon.file-roller-contract</li>
    <li>pantheon.gala</li>
    <li>pantheon.switchboard-plug-bluetooth</li>
    <li>pantheon.switchboard-plug-keyboard</li>
    <li>pantheon.switchboard-plug-mouse-touchpad</li>
    <li>pantheon.switchboard-plug-network</li>
    <li>pantheon.switchboard-plug-pantheon-shell</li>
    <li>pantheon.switchboard-plug-power</li>
    <li>pantheon.switchboard-plug-security-privacy</li>
    <li>pantheon.switchboard-with-plugs</li>
    <li>pantheon.wingpanel</li>
    <li>pantheon.wingpanel-applications-menu</li>
    <li>pantheon.wingpanel-indicator-a11y</li>
    <li>pantheon.wingpanel-indicator-bluetooth</li>
    <li>pantheon.wingpanel-indicator-datetime</li>
    <li>pantheon.wingpanel-indicator-keyboard</li>
    <li>pantheon.wingpanel-indicator-network</li>
    <li>pantheon.wingpanel-indicator-nightlight</li>
    <li>pantheon.wingpanel-indicator-notifications</li>
    <li>pantheon.wingpanel-indicator-power</li>
    <li>pantheon.wingpanel-indicator-sound</li>
    <li>pantheon.wingpanel-quick-settings</li>
    <li>pantheon.wingpanel-with-indicators</li>
    <li>papers</li>
    <li>papers.dev</li>
    <li>papers.devdoc</li>
    <li>phoc</li>
    <li>phosh</li>
    <li>phosh-mobile-settings</li>
    <li>plasma-panel-colorizer</li>
    <li>plasma-panel-spacer-extended</li>
    <li>plasticity</li>
    <li>playonlinux</li>
    <li>power-profiles-daemon</li>
    <li>power-profiles-daemon.devdoc</li>
    <li>protontricks</li>
    <li>protontricks.dist</li>
    <li>protonvpn-cli</li>
    <li>protonvpn-cli.dist</li>
    <li>protonvpn-gui</li>
    <li>protonvpn-gui.dist</li>
    <li>python312Packages.proton-vpn-network-manager</li>
    <li>python312Packages.proton-vpn-network-manager.dist</li>
    <li>python312Packages.protonvpn-nm-lib</li>
    <li>python312Packages.protonvpn-nm-lib.dist</li>
    <li>python312Packages.python-dbusmock</li>
    <li>python312Packages.python-dbusmock.dist</li>
    <li>python312Packages.pywlroots</li>
    <li>python312Packages.pywlroots.dist</li>
    <li>python312Packages.qtile</li>
    <li>python312Packages.qtile-bonsai</li>
    <li>python312Packages.qtile-bonsai.dist</li>
    <li>python312Packages.qtile-extras</li>
    <li>python312Packages.qtile-extras.dist</li>
    <li>python312Packages.qtile.dist</li>
    <li>python313Packages.proton-vpn-network-manager</li>
    <li>python313Packages.proton-vpn-network-manager.dist</li>
    <li>python313Packages.protonvpn-nm-lib</li>
    <li>python313Packages.protonvpn-nm-lib.dist</li>
    <li>python313Packages.python-dbusmock</li>
    <li>python313Packages.python-dbusmock.dist</li>
    <li>python313Packages.pywlroots</li>
    <li>python313Packages.pywlroots.dist</li>
    <li>python313Packages.qtile</li>
    <li>python313Packages.qtile-bonsai</li>
    <li>python313Packages.qtile-bonsai.dist</li>
    <li>python313Packages.qtile-extras</li>
    <li>python313Packages.qtile-extras.dist</li>
    <li>python313Packages.qtile.dist</li>
    <li>qmenumodel</li>
    <li>qmenumodel.dev</li>
    <li>river</li>
    <li>river.man</li>
    <li>rivercarro</li>
    <li>rmenu</li>
    <li>rofi-network-manager</li>
    <li>rofi-vpn</li>
    <li>scenefx</li>
    <li>semantik</li>
    <li>steam</li>
    <li>steam-fhsenv-without-steam</li>
    <li>steam-run</li>
    <li>steam-run-free</li>
    <li>steam-tui</li>
    <li>steamcmd</li>
    <li>streamcontroller</li>
    <li>strongswanNM</li>
    <li>sublime-music</li>
    <li>sublime-music.dist</li>
    <li>sway</li>
    <li>sway-contrib.grimshot</li>
    <li>sway-contrib.grimshot.man</li>
    <li>sway-unwrapped</li>
    <li>swayfx</li>
    <li>swayfx-unwrapped</li>
    <li>systemdgenie</li>
    <li>themechanger</li>
    <li>thermald</li>
    <li>thermald.devdoc</li>
    <li>tinywl</li>
    <li>tokyonight-gtk-theme</li>
    <li>ubports-click</li>
    <li>upower</li>
    <li>upower.dev</li>
    <li>upower.devdoc</li>
    <li>upower.installedTests</li>
    <li>utterly-nord-plasma</li>
    <li>vimix-gtk-themes</li>
    <li>wacomtablet</li>
    <li>waybar</li>
    <li>waybox</li>
    <li>wayfire</li>
    <li>wayfirePlugins.firedecor</li>
    <li>wayfirePlugins.focus-request</li>
    <li>wayfirePlugins.wayfire-plugins-extra</li>
    <li>wayfirePlugins.wayfire-shadows</li>
    <li>wayfirePlugins.wcm</li>
    <li>wayfirePlugins.wf-shell</li>
    <li>wayfirePlugins.windecor</li>
    <li>wayfirePlugins.wwp-switcher</li>
    <li>waywall</li>
    <li>wemeet</li>
    <li>weston</li>
    <li>wifi-qr</li>
    <li>wifi-qr.man</li>
    <li>wingpanel-indicator-ayatana</li>
    <li>wingpanel-indicator-namarupa</li>
    <li>wio</li>
    <li>wlprop</li>
    <li>wlroots_0_17</li>
    <li>wlroots_0_17.examples</li>
    <li>wlroots_0_18</li>
    <li>wlroots_0_18.examples</li>
    <li>wlroots_0_19</li>
    <li>wlroots_0_19.examples</li>
    <li>x11docker</li>
    <li>xdg-desktop-portal</li>
    <li>xdg-desktop-portal-gnome</li>
    <li>xdg-desktop-portal-gtk</li>
    <li>xdg-desktop-portal-hyprland</li>
    <li>xdg-desktop-portal-luminous</li>
    <li>xdg-desktop-portal-shana</li>
    <li>xdg-desktop-portal-termfilechooser</li>
    <li>xdg-desktop-portal-xapp</li>
    <li>xdg-desktop-portal.installedTests</li>
    <li>xfce.xfce4-power-manager</li>
    <li>xfce.xfce4-power-manager.dev</li>
    <li>xfce.xfce4-systemload-plugin</li>
    <li>xfce.xfce4-weather-plugin</li>
    <li>xivlauncher</li>
    <li>xwayland</li>
    <li>xwayland-run</li>
    <li>xwayland-run.man</li>
    <li>xwayland-satellite</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-linux`</summary>
<details>
<summary>libsForQt5.kdev-python</summary>
<pre>/nix/store/sd81bvmch7njdpwx3lkjslixcbj5mivz-python3-3.13.4/include/python3.13/pyconfig.h:24:9: warning: "ALIGNOF_MAX_ALIGN_T" redefined
   24 | #define ALIGNOF_MAX_ALIGN_T 16
      |         ^~~~~~~~~~~~~~~~~~~
/nix/store/sd81bvmch7njdpwx3lkjslixcbj5mivz-python3-3.13.4/include/python3.13/pyport.h:594:12: note: this is the location of the previous definition
  594 | #   define ALIGNOF_MAX_ALIGN_T _Alignof(long double)
      |            ^~~~~~~~~~~~~~~~~~~
/build/kdev-python-23.08.5/parser/astbuilder.cpp: In member function 'Python::CodeAst::Ptr Python::AstBuilder::parse(const QUrl&, QString&)':
/build/kdev-python-23.08.5/parser/astbuilder.cpp:96:5: warning: 'Py_NoSiteFlag' is deprecated [8;;https://gcc.gnu.org/onlinedocs/gcc-14.3.0/gcc/Warning-Options.html#index-Wdeprecated-declarations-Wdeprecated-declarations8;;]
   96 |     Py_NoSiteFlag = 1;
      |     ^~~~~~~~~~~~~
In file included from /nix/store/sd81bvmch7njdpwx3lkjslixcbj5mivz-python3-3.13.4/include/python3.13/Python.h:76,
                 from /build/kdev-python-23.08.5/parser/python_header.h:21:
/nix/store/sd81bvmch7njdpwx3lkjslixcbj5mivz-python3-3.13.4/include/python3.13/cpython/pydebug.h:14:37: note: declared here
   14 | Py_DEPRECATED(3.12) PyAPI_DATA(int) Py_NoSiteFlag;
      |                                     ^~~~~~~~~~~~~
make[2]: *** [parser/CMakeFiles/kdevpythonparser.dir/build.make:169: parser/CMakeFiles/kdevpythonparser.dir/astbuilder.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[2]: *** [parser/CMakeFiles/kdevpythonparser.dir/build.make:183: parser/CMakeFiles/kdevpythonparser.dir/asttransformer.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:732: parser/CMakeFiles/kdevpythonparser.dir/all] Error 2
make: *** [Makefile:136: all] Error 2</pre>
</details>
</details>

mweinelt commented:
When using nixos unstable please subscribe to the unstable breaking changes thread on Discourse.

https://discourse.nixos.org/t/breaking-changes-announcement-for-unstable/17574/85

For everyone else I added a prominent note to the release notes.

We cannot really influence how the various upstream UIs for NM display errors I'm afraid.

mweinelt commented on nixos/modules/services/networking/networkmanager.nix:
```diff
@@ -220,30 +225,37 @@ in
-          List of NetworkManager plug-ins to enable.
-          Some plug-ins are enabled by the NetworkManager module by default.
+        example = literalExpression ''
+          [
```
Yes, it should. https://github.com/NixOS/nixpkgs/pull/479104
